e2-studio-star-rx72n-firmware/tests/mocks/mock_rx_hcsr04_irq.c

```
@@ -0,0 +1,447 @@
+ * @note Validates parameters to match real API error paths (Liskov substitution)
+ * @note Thread-safe: no shared state
+ * @code
+ * rx_err_t err = rx_hcsr04_isr_register((uint8_t)k_hcsr04_irq_11,
+ *                                        k_hcsr04_sensor_front_left);
+ * assert(err == k_rx_ok);
+ * @endcode
+ *
+ * @see rx_hcsr04_isr_register() Real implementation in rx_hcsr04_isr.h
+ * @since Version 1.2.0 (Issue #296)
+ */
+rx_err_t rx_hcsr04_isr_register(const uint8_t irq_num, const rx_hcsr04_sensor_index_t sensor_index)
```

Clang (14.0.6)

[warning] 2 adjacent parameters of 'rx_hcsr04_isr_register' of convertible types are easily swapped by mistake

(bugprone-easily-swappable-parameters)